### ISSUE TITLE
Toolbox with categories takes focus but doesn't indicate it or have working nav

### ISSUE THREAD

**Issue by microbit-matt-hillsdon** [rachel-fenichel/blockly-keyboard-experimentation#149]
https://category-toolbox.blockly-keyboard-experimentation.pages.dev/ is this plugin [but with the toolbox modified to have categories](https://github.com/google/blockly-keyboard-experimentation/compare/main...microbit-matt-hillsdon:blockly-keyboard-experimentation:category-toolbox?expand=1).

The toolbox is currently an invisible tab position before the workspace. When focussed keyboard navigation doesn't work (as the workspace is not focussed). The navigation works when starting the interaction via the workspace.

I think this should be a real tab position with a focus indicator and working keyboard navigation when focussed, but would be good to get @kmcnaught's input here too. Complicated for now because focus has to be on the workspace for keyboard navigation to work in the toolbox.

Please can we also consider enhancing the demo to have a toolbox variant and more varied blocks? Or is there some other testing environment we can use to test the toolbox nav?

**Comment by kmcnaught** [comment]
I'm pretty sure that it is indeed intended (but not yet implemented) that the toolbox is a _real tab position with a focus indicator and working keyboard navigation when focussed_

See [discussion here](https://github.com/google/blockly-keyboard-experimentation/issues/101)

I agree with the ask for a category toolbox, but might not be worth it until the tab/nav behaviour is there.

**Comment by rachel-fenichel** [comment]
@BenHenning fyi

**Comment by BenHenning** [comment]
I need to test how the multiple categories is affected by #225.

**Comment by BenHenning** [comment]
So I've tested the demo changes you've kindly provided @microbit-matt-hillsdon with the changes from #225. I think enabling keyboard navigation when stopped at the toolbox makes the subcategory experience better, but there are definitely some issues. For instance, a category with a subcategory doesn't seem to handle right arrow correctly as that opens the subcategory rather than navigating to the category's direct blocks (and, in fact, there's no way to navigate to them).

Separately, I've noticed that navigating left to collapse a category causes the flyout to disappear. There's definitely work we can do here to make this better:

https://github.com/user-attachments/assets/90b89b0d-0242-4863-8414-b4383b68c8e5

I think for now #225 does seem to help with this issue, and I'll also update that PR to add subcategories to the demo so that we can iterate on that. There will be more to do once #225 is merged, but this seems like a reasonable starting place.
